### PR TITLE
Background approval

### PR DESCRIPTION
**Pull request by jkeat**
Links
-----
- Dependency PRs: https://github.com/UPchieve/server/pull/376

Description
-----------
- For regular volunteers, admin approval only happens once background info has been completed
- For partner volunteers, auto-approve on background info completion

todo (in a future PR)
- send a tweaked version of the "approved but not onboarded" email to partner volunteers on background info submission

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: tests/unit/services/UserService.test.ts [hunk 1]
```diff
@@ -348,7 +348,7 @@ test('Open volunteer is not approved when submitting their background info is no
   expect(accountApprovedUserAction).toBeNull();
 });
 
-test('Open volunteer is approved when submitting their background info is the final approval step', async () => {
+test('Open volunteer is not approved when submitting their background info is the final approval step', async () => {
```
Comment (jkeat) [inline tests/unit/services/UserService.test.ts:351]: @treykris this test is a bit redundant now, do you think it'd be better to remove or just keep it like this?

Comment (treykris) [reply tests/unit/services/UserService.test.ts:351]: Yup I agree, I think it's better to remove it
